File: app/src/main/java/com/br/rafael/TesteAndroidv2/data/model/StatementResponse.java
package com.br.rafael.TesteAndroidv2.data.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class StatementResponse implements Parcelable {

    @SerializedName("statementList")
    private ArrayList<Statement> statementArrayList;

    public StatementResponse() {

    }

    protected StatementResponse(Parcel in) {
        statementArrayList = in.createTypedArrayList(Statement.CREATOR);
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(statementArrayList);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<StatementResponse> CREATOR = new Creator<StatementResponse>() {
        @Override
        public StatementResponse createFromParcel(Parcel in) {
            return new StatementResponse(in);
        }

        @Override
        public StatementResponse[] newArray(int size) {
            return new StatementResponse[size];
        }
    };

    public ArrayList<Statement> getStatementArrayList() {
        return statementArrayList;
    }
}

File: BankApp/app/src/main/java/com/lucianogiardino/bankapp/domain/model/StatementResponse.kt
package com.lucianogiardino.bankapp.domain.model

import com.google.gson.annotations.SerializedName
import com.lucianogiardino.bankapp.domain.model.Error
import com.lucianogiardino.bankapp.domain.model.Statement

data class StatementResponse(

    @SerializedName("statementList")
    val statementList : ArrayList<Statement>,

    @SerializedName("error")
    val error : Error

)